Latest master segfaults immediately (Bisected).

**Describe the bug**

Latest master immediately segfaults after starting.

**To Reproduce**
1. Start latest darktable development version.

**Expected behaviour**
To start nice and proper.

**Bisect info**

```
7c4dbe72a4519b39b75976a0d2553b3b26292dac is the first bad commit
commit 7c4dbe72a4519b39b75976a0d2553b3b26292dac
Author: ralfbrown <ralfbrown@users.noreply.github.com>
Date:   Thu Mar 4 01:03:24 2021 -0500

    apply changes requested in review

:040000 040000 bfc924d1fea09fa4456c15e5544d08a9aa76470a 9f0df341b7ae0cea03a5952cbe3c09ced41bad93 M      src
```

Ping: @ralfbrown 

**Version**

* darktable version : 3.5.0+1340 ge448e6d6d / 3.5.0+1344~g597a63805

**Platform**

* OS                : Linux (kernel 5.9.15-1)
* Distro            : Debian Buster (rel. 10.8)
* Graphics driver   : nvidia (460.39)
* OpenCL installed  : Yes (nvidia-opencl-common 460.39-1)
* OpenCL activated  : Yes
* Xorg              : 7.7+19

**Build info**

* Compiler      : gcc 8.3.0
* Cmake         : 3.13.4
* Glib          : 2.58.3-2
* Gtk+          : 3.24.5-1
* Exiv2         : 0.27.99.0
* Sqlite        : 3.27.2
* Optimization  : None

**Additional info**

Crashlog: [darktable_bt_QBR5Z0.txt](https://github.com/darktable-org/darktable/files/6095505/darktable_bt_QBR5Z0.txt)

this is darktable 3.5.0+1344~g597a63805
copyright (c) 2009-2021 johannes hanika
darktable-dev@lists.darktable.org

compile options:
  bit depth is 64 bit
  normal build
  SSE2 optimized codepath enabled
  OpenMP support enabled
  OpenCL support enabled
  Lua support enabled, API version 6.2.3
  Colord support enabled
  gPhoto2 support enabled
  GraphicsMagick support enabled
  ImageMagick support disabled
  OpenEXR support enabled

Changed the line you pointed out and started the build process again.


Build stops with this error:

```
[ 59%] Built target collect
/home/jade/.local/git/darktable/src/libs/export.c:1056:13: error: '_destroy_child' defined but not used [-Werror=unused-function]
 static void _destroy_child(GtkWidget *widget, gpointer data)
```

Full build log [dt.build.2021.03.06.1729.log](https://github.com/darktable-org/darktable/files/6095688/dt.build.2021.03.06.1729.log)

> ```
> /home/jade/.local/git/darktable/src/libs/export.c:1056:13: error: '_destroy_child' defined but not used [-Werror=unused-function]
> ```

You would need to surround the _destroy_child function with `#if 0` and `#endif` for a succesful compile with `#if 0 //TODO`.  Although you narrowed the issue down to four Lua plugins, it still sounds like an interaction between something in those files and the change in the C code.

--- a/src/libs/export.c
+++ b/src/libs/export.c
@@ -1053,10 +1053,10 @@ static void _update_formats_combobox(dt_lib_export_t *d)
   gtk_widget_set_sensitive(d->format, !empty);
 }
 
-static void _destroy_child(GtkWidget *widget, gpointer data)
+static void _remove_child(GtkWidget *widget, gpointer data)
 {
-  (void)data;  // avoid unreferenced-parameter warning
-  gtk_widget_destroy(widget);
+  GtkContainer *cont = (GtkContainer *)data;
+  gtk_container_remove(cont, widget);
 }
 
 static void _on_storage_list_changed(gpointer instance, dt_lib_module_t *self)
@@ -1066,7 +1066,8 @@ static void _on_storage_list_changed(gpointer instance, dt_lib_module_t *self)
   dt_bauhaus_combobox_clear(d->storage);
 
 #if 1  //TODO: this code needs careful testing before removing the old code in the 'else' branch
-  gtk_container_foreach(GTK_CONTAINER(d->storage_extra_container), _destroy_child, NULL);
+  GtkContainer *cont = GTK_CONTAINER(d->storage_extra_container);
+  gtk_container_foreach(cont, _remove_child, cont);
 #else //old code below
   GList *children = gtk_container_get_children(GTK_CONTAINER(d->storage_extra_container));
   for(GList *iter = children; iter; iter = g_list_next(iter))

